uploader/test/test_file_upload.py

```diff
@@ -0,0 +1,193 @@
+import logging
+import os
+import tempfile
+import zipfile
+from typing import Dict, List
+
+from django.test import TestCase
+from django.utils import timezone
+from openpyxl.workbook import Workbook
+
+from core.management.commands.data_migration import data_migration, _val_handler_person__organisation_type, \
+    _val_handler_empty_str_null
+from core.models import Iso639LanguageCode, SEQ_NAME_ISO_LANGUAGE__LANGUAGE_ID, CofkUnionOrgType
+from institution.models import CofkUnionInstitution
+from location.models import CofkUnionLocation
+from person.models import SEQ_NAME_COFKUNIONPERSION__IPERSON_ID, CofkUnionPerson
+from uploader.constants import MANDATORY_SHEETS
+from uploader.models import CofkCollectUpload, CofkCollectStatus
+from uploader.spreadsheet import CofkUploadExcelFile
+from uploader.validation import CofkExcelFileError
+
+log = logging.getLogger(__name__)
+
+
+def create_excel_file(data: Dict[str, List[List]]=None) -> str:
+    wb = Workbook()
+    tf = tempfile.NamedTemporaryFile(suffix='.xlsx', delete=False)
+
+    for sheet_name in MANDATORY_SHEETS.keys():
+        ws = wb.create_sheet(sheet_name)
+        column_count = len(MANDATORY_SHEETS[sheet_name]['columns'])
+        ws.append(MANDATORY_SHEETS[sheet_name]['columns'])
+        ws.append(['-'] * column_count)
+
+        if data and sheet_name in data:
+            for row in data[sheet_name]:
+                ws.append(row)
+
+    wb.save(tf.name)
+
+    return tf.name
+
+def migrate(model, params=None):
+    data_migration(**{'user': os.getenv('POSTGRES_USER'),
+                      'password': os.getenv('POSTGRES_PASSWORD'),
+                      'host': os.getenv('POSTGRES_HOST'), 'port': 5432,
+                      'database': 'ouls', 'model': model,
+                      'params': params})
+
+class TestFileUpload(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        migrate(CofkCollectStatus)
+        migrate(CofkUnionOrgType)
+        migrate(Iso639LanguageCode, {'seq_name': SEQ_NAME_ISO_LANGUAGE__LANGUAGE_ID,
+                                     'int_pk_col_name': 'language_id'})
+        migrate(
+            CofkUnionPerson, {
+            'col_val_handler_fn_list': [_val_handler_person__organisation_type, ],
+            'seq_name': SEQ_NAME_COFKUNIONPERSION__IPERSON_ID,
+            'int_pk_col_name': 'iperson_id',}
+        )
+        migrate(CofkUnionLocation)
+        migrate(CofkUnionInstitution,
+                {'col_val_handler_fn_list': [_val_handler_empty_str_null]})
```

Comment:
Sorry, the `data_migration` and old db should be no longer used after first time deployment.

you should create fixtures records.

if you really need all data from table, you might need to export data from new DB to sql file